tests/rptest/remote_scripts/stream_verifier_txn.py

```diff
@@ -0,0 +1,1639 @@
+        """
+        self.logger.info("Start of consuming messages")
+        self.consume_thread = self.create_thread(
+            self._consume_from_topic, thread_name="stream_consume_thread")
+        if wait:
+            self.consume_thread.join()
+        return
+
+    @staticmethod
+    def _consume_from_topic(logger: logging.Logger,
+                            topic: TopicStatus) -> TopicStatus:
+        """Consumes all messages from topic
+
+        Args:
+            logger (logging.Logger): logger class
+            topic (TopicStatus): current live topic status class to work with
+
+        Raises:
+            ck.KafkaException: on any error occured while consuming
+
+        Returns:
+            int: number of consumed messages
+        """
+        def time_since_last_msg() -> int:
+            diff_ms = datetime.now().timestamp() - topic.last_message_ts
+            return int(diff_ms * 1000)
+
+        if topic.reached_eof:
+            # Just skip it, nothing consumed
+            logger.debug(f"...topic {topic.source_topic_name} already at eof")
+            return topic
+
+        # Message consuming loop
+        consumer = ck.Consumer(topic.consumer_config)
+        try:
+            # Recent changes offers use of subscribe
+            # regardless of transactions and partitions
+            consumer.subscribe([topic.source_topic_name])
+            topic.last_message_ts = datetime.now().timestamp()
+            while True:
+                # calculate elapsed time
+                _since_last_msg_ms = time_since_last_msg()
+
+                # Exit on timeout
+                if _since_last_msg_ms > topic.consume_timeout_s:
+                    logger.error("Timeout consuming messages "
+                                 f"from {topic.source_topic_name}")
+                    break
+
+                # Poll for the message
+                msg = consumer.poll(timeout=app_config.consume_poll_timeout)
+                if msg is None:
+                    # no messages
+                    continue
+                # On error, check for the EOF
+                if msg.error():
+                    if msg.error().code() == ck.KafkaError._PARTITION_EOF:
+                        # End of partition event
+                        logger.info(f"Consumer of '{msg.topic()}' "
+                                    f"[{msg.partition()}] reached "
+                                    f"end at offset {msg.offset()}")
+                        topic.reached_eof = True
+                        break
+                    # If not EOF, raise the error
+                    elif msg.error():
+                        raise ck.KafkaException(msg.error())
+                else:
+                    topic.last_message_ts = datetime.now().timestamp()
+                    # Increment index
+                    topic.index += 1
+                    # For consuming operation, processed count
+                    # almost always equals to index
```

Comment:
I thought this would validate for atomicity or some such? (monotonicity of integers)

Reply:
Looks like validation part is missing here. I should add it.